### ISSUE TITLE
Error creating MongoDB Cluster

### ISSUE THREAD

**Issue by deepak7093** [mongodb/terraform-provider-mongodbatlas#9]
Error: error creating MongoDB Cluster: POST https://cloud.mongodb.com/api/atlas/v1.0/groups/xxxxxx/clusters: 500 (request "Internal Server Error") Unexpected error.

Seems like API servers had issue.

terraform version: v0.12.3
Atlas provider version: 0.1.0
terraform configuration:  https://gist.github.com/deepak7093/c286f09316142d424832148f36d47c4a
Debug Logs
```
2019/08/22 18:49:29 [DEBUG] mongodbatlas_cluster.clusters-copilot[0]: apply errored, but we're indicating that via the Error pointer rather than returning it: error creating MongoDB Cluster: POST https://cloud.mongodb.com/api/atlas/v1.0/groups/xxxxxx/clusters: 500 (request "Internal Server Error") Unexpected error.
2019/08/22 18:49:29 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2019/08/22 18:49:29 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: error creating MongoDB Cluster: POST https://cloud.mongodb.com/api/atlas/v1.0/groups/xxxx/clusters: 500 (request "Internal Server Error") Unexpected error.
2019/08/22 18:49:29 [ERROR] <root>: eval: *terraform.EvalSequence, err: error creating MongoDB Cluster: POST https://cloud.mongodb.com/api/atlas/v1.0/groups/xxxxxx/clusters: 500 (request "Internal Server Error") Unexpected error.
2019/08/22 18:49:29 [TRACE] [walkApply] Exiting eval tree: mongodbatlas_cluster.clusters-copilot[0]
2019/08/22 18:49:29 [TRACE] vertex "mongodbatlas_cluster.clusters-copilot[0]": visit complete
2019-08-22T18:49:29.324+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4: 2019/08/22 18:49:29 [DEBUG] MongoDB Atlas API Response Details:
2019-08-22T18:49:29.324+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4: ---[ RESPONSE ]--------------------------------------
2019-08-22T18:49:29.324+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4: HTTP/2.0 500 Internal Server Error
2019-08-22T18:49:29.324+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4: Content-Length: 122
2019-08-22T18:49:29.324+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4: Content-Type: application/json
2019-08-22T18:49:29.324+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4: Date: Thu, 22 Aug 2019 13:19:29 GMT
2019-08-22T18:49:29.324+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4: Strict-Transport-Security: max-age=300
2019-08-22T18:49:29.324+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4:
2019-08-22T18:49:29.324+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4: {
2019-08-22T18:49:29.324+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4:  "detail": "Unexpected error.",
2019-08-22T18:49:29.324+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4:  "error": 500,
2019-08-22T18:49:29.324+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4:  "errorCode": "UNEXPECTED_ERROR",
2019-08-22T18:49:29.325+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4:  "parameters": [],
2019-08-22T18:49:29.325+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4:  "reason": "Internal Server Error"
2019-08-22T18:49:29.325+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4: }
2019-08-22T18:49:29.325+0530 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.1.0_x4: -----------------------------------------------------
2019/08/22 18:49:29 [DEBUG] mongodbatlas_cluster.clusters-fabric[0]: apply errored, but we're indicating that via the Error pointer rather than returning it: error creating MongoDB Cluster: POST https://cloud.mongodb.com/api/atlas/v1.0/groups/xxxx/clusters: 500 (request "Internal Server Error") Unexpected error.
2019/08/22 18:49:29 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2019/08/22 18:49:29 [TRACE] EvalMaybeTainted: mongodbatlas_cluster.clusters-fabric[0] encountered an error during creation, so it is now marked as tainted

Error: error creating MongoDB Cluster: POST https://cloud.mongodb.com/api/atlas/v1.0/groups/xxxxxx/clusters: 500 (request "Internal Server Error") Unexpected error.
```

**Comment by deepak7093** [comment]
It seems an issue with M2 instance size. I am able to create a cluster with M10.

**Comment by PacoDw** [comment]
Hello @deepak7093 , this issue was resolved with the pr #11 , you can test it on the `fix-#2` branch.

Let us know if you have any comments, thanks!
